File: library/src/main/java/com/snail/easyble/callback/RequestCallback.java
package com.snail.easyble.callback;

import android.support.annotation.NonNull;

import com.snail.easyble.annotation.InvokeThread;
import com.snail.easyble.event.Events;

/**
 * 描述: 请求回调
 * 时间: 2018/12/2 10:32
 * 作者: zengfansheng
 */
public interface RequestCallback<T> {
    @InvokeThread
    void onSuccess(@NonNull T data);

    @InvokeThread
    void onFail(@NonNull Events.RequestFailed requestFailed);
}

File: library/src/main/java/com/snail/easyble/callback/RequestCallback.kt
package com.snail.easyble.callback

import com.snail.easyble.annotation.InvokeThread
import com.snail.easyble.event.Events

/**
 * date: 2018/12/2 10:32
 * author: zengfansheng
 */
interface RequestCallback<T> {
    @InvokeThread
    fun onSuccess(data: T)

    @InvokeThread
    fun onFail(requestFailed: Events.RequestFailed)
}
